fastqheat/__main__.py

```diff
@@ -17,7 +17,15 @@
 
 SRR_PATTERN = re.compile(r'^(SRR|ERR|DRR)\d+$')
 SRP_PATTERN = re.compile(r'^(((SR|ER|DR)[PAXS])|(SAM(N|EA|D))|PRJ(NA|EB|DB)|(GS[EM]))\d+$')
-USABLE_CPUS_COUNT = len(os.sched_getaffinity(0))
+
+try:
+    USABLE_CPUS_COUNT = len(os.sched_getaffinity(0))
+except AttributeError:
+    # sched_getaffinity is not available on this OS, fall back to cpu_count
+    if (cpu_count := os.cpu_count()) is not None:
+        USABLE_CPUS_COUNT = cpu_count
+    else:
+        USABLE_CPUS_COUNT = 4
```

Comment:
Seems like a reasonable default for the number of threads if we're unable to determine the core count? What do you suggest?

Reply:
@dakq I agree with that 4 is reasonable. However, what will happen, if user has less than 4 threads/cores?

Reply:
@dakq I believe the safest way would be not to pass number of cores argument to fastrq_dump and pigz at all if this number wasn't explicitly specified by a user

Reply:
> what will happen, if user has less than 4 threads/cores?

The user's machine will have to deal with 2-3 more threads than the number of cores they have. Do you think this will cause any actual problems?

> I believe the safest way would be not to pass number of cores argument to fastrq_dump and pigz at all if this number wasn't explicitly specified by a user

pigz uses 8 compression threads by default. fasterq-dump uses 6.

Reply:
| pigz uses 8 compression threads by default. fasterq-dump uses 6.

@dakq in this case 4  cores feels safe, thank you for the clarification. Please, let the comment about it next to defining the default numbers of cores. Something like

```python
def get_cpu_cores_count() -> int:
    try:
        return len(os.sched_getaffinity(0))
    except AttributeError:
        # sched_getaffinity is not available on this OS, fall back to cpu_count
        if (cpu_count := os.cpu_count()) is not None:
            return cpu_count
        else:
            # For now we only use it to pass to fasteq-dump and to pigz external utilities.
            # pigz uses 8 threads by default. fasterq-dump uses 6, so 4 feels like a nice safe option
            return 4
```

FYI https://www.codebyamir.com/blog/software-anti-patterns-magic-numbers